ffmpeg/ffmpeg.go

```diff
@@ -506,11 +506,14 @@ func (t *Transcoder) Transcode(input *TranscodeOptionsIn, ps []TranscodeOptions)
 		t.started = true
 	}
 	if !t.started {
-		status, _, vcodec, _, _ := GetCodecInfo(input.Fname)
-		// TODO: Check following condition, is vcodec == "" ?
-		videoMissing := status == CodecStatusNeedsBypass || vcodec == ""
-		if videoMissing {
-			// Audio-only segment, fail fast right here as we cannot handle them nicely
+		status, _, _, _, _ := GetCodecInfo(input.Fname)
+		// DONE: Check following condition, is vcodec == "" ?
+		// videoMissing := status == CodecStatusNeedsBypass || vcodec == ""
+		// if videoMissing {
+		if status == CodecStatusNeedsBypass {
+			// > no longer valid comment > Audio-only segment, fail fast right here as we cannot handle them nicely
```

Comment:
Hi @darkdarkdragon @cyberj0g, I need help on this change. 
The comment initially made sense but now regression was detected in `go-livepeer/core/transcoder_test.go::TestAudioCopy()`.

Its not clear to me what exactly `NeedsBypass` signals in the input media file? Depending on the answer we should keep change of this condition or change how tests use transcoder for audio only files.

Here are the links to original logic:
- [C function returning NeedsBypass](https://github.com/livepeer/lpms/blob/master/ffmpeg/extras.c#L156)
- [go function returning bool](https://github.com/livepeer/lpms/blob/master/ffmpeg/ffmpeg.go#L125)
- [original condition](https://github.com/livepeer/lpms/blob/master/ffmpeg/ffmpeg.go#L388)

With my interpretation being: returning error in case of audio-only segment. [`ErrTranscoderVid = errors.New("TranscoderInvalidVideo")`](https://github.com/livepeer/lpms/blob/master/ffmpeg/ffmpeg.go#L33)

Should we follow the intent of the comment `Audio-only segment, fail fast right here as we cannot handle them nicely` ? Or keep original logic and update comment appropriately?

Reply:
From quirks.md:

We bypass the usual transcoding process for audio-only segments and just return them back as-is. To do this, we can check if a valid video stream is present without any actual video frames, and if so we skip transcoding.
While [this check](https://github.com/livepeer/lpms/blob/fe330766146dba62f3e1fccd07a4b96fa1abcf4d/ffmpeg/extras.c#L110-L117) is still implemented in LPMS, and was originally used by Transcoders, now this function is directly used by the Broadcaster since https://github.com/livepeer/go-livepeer/pull/1933.

In other words, audio-only segment shouldn't make its way into transcoder. The corresponding test should just check for correct error message.